packages/internet-header/src/components/post-internet-header/post-internet-header.tsx

```diff
@@ -152,10 +152,28 @@ export class PostInternetHeader {
   private updateLogoAnimation: () => void;
   private scrollParent: Element | Document;
 
+  private isValidVersion(str: string): boolean {
+    if (typeof str !== 'string') return false;
+    const validCharRegex = /^[a-z0-9-]+$/;
+    return validCharRegex.test(str);
+  }
+
   constructor() {
     if (this.project === undefined || this.project === '' || !isValidProjectId(this.project)) {
       throw new Error(
-        `Internet Header project key is "${this.project}". Please provide a valid project key.`,
+        `Internet Header project key is "${this.project}". Please provide a valid project key.`
+      );
+    }
+    
+    const validEnvironments = ['prod', 'int01', 'dev'];
```

Comment:
These are not all valid environments. Have a look at https://github.com/swisspost/design-system/blob/main/packages/internet-header/src/models/general.model.ts#L39.